manifests/apache.pp

```diff
@@ -1,34 +1,133 @@
 # Configure an Apache vhost
 # @api private
-class pulpcore::apache {
+class pulpcore::apache (
+  Boolean $manage_selinux_boolean = true,
+  Stdlib::Port $http_port = 80,
+  Stdlib::Port $https_port = 443,
+  Hash[String, Any] $http_vhost_options = {},
+  Hash[String, Any] $https_vhost_options = {},
+  Enum['none', 'optional', 'require', 'optional_no_ca'] $ssl_verify_client = 'optional',
+) {
+  $vhost_priority = $pulpcore::apache_vhost_priority
```

Comment:
Should this be a param on this class? I am still not clear in what cases you want to tie it to the modules variables and when you want to make it available as a param on the class to pass in when it's a private API.

Reply:
I don't have a hard rule on when I make it a class parameter or inside the class. What's important to me is how coupled things are. If you use a lot from the other class and the private class ends up being unusable without the main class, it may not worth it. If there's a few, you can end up in a situation where you can run the rspec tests on just the private class without including the main class. That means it's smaller to compile and thus faster tests.

In this particular case I think I started with hardcoding it but later realized it should be a proper parameter.